rero/rero-ils
# patrons: add new patron fields

* Adds notes to user resource.
* Adds expiration date in the patron informations.
* Adds libraries affilation in the patron informations.
* Removes duplicates patron custom validation.
* Groups all patron informations in a nested patron structure.
* Cleans patron tests.
* Displays expiration date and notes in the patron profile.
* Adds web API to return the number of patrons given a username.

Signed-off-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Dependencies:

- rero/rero-ils-ui#385

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1590?milestone=273464

## How to test?

- Login with simonetta with the username instead of the email.
- Check in the profile the note below the tabs and the presence of the expiration date in the personal information.

## Code review check list
